src/test.core.c

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#include <stdio.h>
+
+// Include all headers.
+#include <openassetio/c/InfoDictionary.h>
+#include <openassetio/c/StringView.h>
+#include <openassetio/c/errors.h>
+#include <openassetio/c/hostApi/Manager.h>
+#include <openassetio/c/managerApi/CManagerInterface.h>
+#include <openassetio/c/managerApi/HostSession.h>
+#include <openassetio/c/managerApi/ManagerInterface.h>
+#include <openassetio/c/namespace.h>
```

Comment:
Just an observation from the outside world... you've got almost as many lines of #includes as there are in the body of main.

Are there thoughts about a utility header "#include <openassetio/c/openassetio.h>" for example that brings them all in?

Reply:
This is specifically to test that all expected headers exist and can compile (in this case in-particular with a C compiler). We wouldn't expect users to include all the headers in normal usage.

Having said that, it would be handy to ease maintenance of this repo if OpenAssetIO maintained an uber header instead :thinking:.

Reply:
> Having said that, it would be handy to ease maintenance of this repo if OpenAssetIO maintained an uber header instead

Would users also find it useful? is there any visibility on whether they're regularly #include'ing the majority of the exposed public headers?


Q: Do you expect all headers to each be standalone?

In a previous company, we used to have tests that autogenerated a c file per header, which just had a single #include for its corresponding header, and was compiled (not linked, just compiled).

It then confirmed whether that header had any missing #includes itself, or forward declarations, to avoid consumers needing to know about them.

Reply:
Of course. It's out of scope for this PR.

Reply:
Added https://github.com/OpenAssetIO/OpenAssetIO-Test-CMake/issues/2